### PR TITLE
Report zero values in HVAC Sizing Summary with zero loads

### LINKED ISSUE #8124: Unexpected behavior related to HVAC Sizing Summary: Zone Sensible Cooling table
Issue overview
--------------
**Observed Behavior**:  When simulating the attached file "**in-fails.idf**", the E+ "**HVAC Sizing Summary**" **Zone Sensible Cooling** table in the HTML output **fails to populate with ANY data**, which is causing issues with downstream SQL queries that retrieve data from this view , and which is post-processed by external tools.

image

**Desired Behavior**: When simulating using the file "**in-fails.idf**", the E+ "**HVAC Sizing Summary**" **Zone Sensible Cooling** table in the HTML output (and the SQLite tablular file used to render it) should populate with non - empty values. 

[in-works.zip](https://github.com/NREL/EnergyPlus/files/4854476/in-works.zip)
[in-fails.zip](https://github.com/NREL/EnergyPlus/files/4854477/in-fails.zip)

**Details**: 
To reproduce the issue, zip files including (2) idf files have been provided:

1. in-works.idf
2. in-fails.idf

In addition, the .err files and .html output reports generated from simulating the idf files, have been included in each zipfile. 

The "in-fails.err" file generated from "in-fails.idf" suggests (in lines 47/48) the following:

   **** Warning ** Calculated design cooling load for zone=LIVING SPACE is zero.
   **   ~~~   ** Check Sizing:Zone and ZoneControl:Thermostat inputs.**

Inspecting "in-fails.idf", It is unclear to the user how the "calculated design load" for the living space thermal zone would be == 0.00. To this end, the "in-fails.idf" file has been carefully inspected - the "living space" thermal zone has been confirmed to have a valid cooling design day description, skin (envelope) loads, a dual setpoint thermostat object configured properly for cooling, including "summer design day" schedules, and internal gains (lights/equipment) with cooling design day values schedule set to 1.0. At a minimum, the user believes the "skin loads" should generate a non zero design cooling load. A value of 0 seems like a bug.   

As a comparative test, the "in-works.idf" file contains much of the same physical description of the "in-fails.idf" file,  - including IDENTICAL Sizing:Zone and ZoneControl:Thermostat objects.  Changes between the two .idf files include materials properties and construction, and thus reduced HVAC hard sized values, but nothing that would substantiate the observed behavior. The "in.works.idf file **does not produce the design cooling load warning**, and the tabular values in the HVAC Sizing Summary: Zone Sensible Cooling table for this .idf file are non zero. 

In both idf files, the Coil:Cooling:DX:SingleSpeed object (* a child of a AirLoopHVAC:UnitarySystem object) has it's capacity "hard sized". 

Some additional details for this issue : Windows 10 64 Bit 
 - Version of EnergyPlus: 9.3

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
Report zero values in the HVAC Sizing Summary Zone Sensible Cooling and Heating tables for every conditioned zone, even when there is zero cooling or heating load. Previously zones with a zero load were left out of the table.

Fixes #8124.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] ~If feature, test running new feature, try creative ways to break it~
 - [ ] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] ~Check any new function arguments for performance impacts~
 - [ ] ~Verify IDF naming conventions and styles, memos and notes and defaults~
 - [ ] ~If new idf included, locally check the err file and other outputs~

### CODE REVIEW

**Comment by lgu1234** [comment]
https://github.com/chrisbalbach I uploaded an initial fix to report zero values with zero cooling load to avoid empty values in the table. Please take a look and let me know your comments. If OK, I am going to add a unit test.

**Comment by lgu1234** [comment]
Added a screen shot:
New

**Comment by lgu1234** [comment]
https://github.com/mdahlhausen, https://github.com/chrisbalbach, @mjwitte 
I debug the program at 3PM in the summer design day during sizing to see which components are contributed to zero cooling load.

After CalcZoneSums called by the Predictor, I found no internal gain, low heat transfer rate between surfaces and zone (Living Space), and minor infiltration. 

The low heat transfer rate is caused by low interior surface temperature of slab at 21C (23.89 zone setpoitn) from surface convection gain. That is why no cooling load. Here is current slab construction:

Construction,
  Slab construction,                      !- Name
  Concrete 4.0 in.,                       !- Layer 1
  Concrete 4.0 in. 1;                     !- Layer 2

I think concrete exposure at interior surface may not be realistic. When I tried to change the construction with carpet coverage:

Construction,
  Slab construction,                      !- Name
  Concrete 4.0 in.,                       !- Layer 1
  CP02;                     !- Layer 2

! Carpet With Rubber Pad

  Material:NoMass,
    CP02,                    !- Name
    Rough,                   !- Roughness
    0.2170000,               !- Thermal Resistance {m2-K/W}
    0.9000000,               !- Thermal Absorptance
    0.7500000,               !- Solar Absorptance
    0.7500000;               !- Visible Absorptance

I got a fatal message:

   ** Severe  ** Construction="SLAB CONSTRUCTION", constructions referenced by surfaces with a
   **   ~~~   ** "Foundation" Outside Boundary Condition must use only regular material objects
   **   ~~~   ** Material="CP02", is not a regular material object
   **  Fatal  ** GetSurfaceData: Errors discovered, program terminates.

In general, the program does it as expected. Please let me know your concerns.

**Comment by mjwitte** [comment]
What is the outside boundary condition of the slab? If "Ground" what are the GroundTemperature:BuildingSurface values?

**Comment by mjwitte** [comment]
Oh, I see, it's using "Foundation". Perhaps look at the inputs for that.

**Comment by lgu1234** [comment]
@mjwitte The foundation type is Kiva. The value of TempSurfOut for the slab is 0. Kiva performs its own calculation.

**Comment by lgu1234** [comment]
@mjwitte In addition, I looked at values of SumHA and SumHATsurf from output of CalcZoneSums: 7341.96 and 173526.2, so that the equivalent surface temperature is 23.63487, which is below zone setpoint at 23.89. That is why no cooling load.


 
-- | --

**Comment by lgu1234** [comment]
The table big diffs occur in the ASHRAE9012016_HotelSmall_Denver, because zero values are reported in MechanicalRoomFLR1 in both Zone Sensible Heating and Zone Sensible Cooling under Report: HVAC Sizing Summary. The diffs are expected.

File: src/EnergyPlus/SizingManager.cc [hunk 1]
```diff
@@ -661,6 +661,24 @@ namespace SizingManager {
                     PreDefTableEntry(pdchZnClPkOAHum, curName, HumRatAtPeak, 5);
                     PreDefTableEntry(pdchZnClPkOAMinFlow, curName, FinalZoneSizing(CtrlZoneNum).MinOA, 3);
                     PreDefTableEntry(pdchZnClPkDOASHeatGain, curName, DOASHeatGainRateAtClPk);
+                } else {
+                    curName = FinalZoneSizing(CtrlZoneNum).ZoneName;
+                    PreDefTableEntry(pdchZnClCalcDesLd, curName, 0.0);
+                    PreDefTableEntry(pdchZnClUserDesLd, curName, 0.0);
+                    if (Zone(ZoneNum).FloorArea != 0.0) {
```
Comment (mjwitte) [inline src/EnergyPlus/SizingManager.cc]: Nit picking - this `if` is not necessary, just always write 0.0 here. Same comment in the heating section.

File: src/EnergyPlus/SizingManager.cc [hunk 2]
```diff
@@ -661,6 +661,24 @@ namespace SizingManager {
...
+                    PreDefTableEntry(pdchZnClDesDay, curName, "N/A");
+                    PreDefTableEntry(pdchZnClPkTime, curName, "N/A");
+                    PreDefTableEntry(pdchZnClPkTstatTemp, curName, 0.0);
+                    PreDefTableEntry(pdchZnClPkIndTemp, curName, 0.0);
+                    PreDefTableEntry(pdchZnClPkIndHum, curName, 0.0, 5);
+                    PreDefTableEntry(pdchZnClPkOATemp, curName, 0.0);
+                    PreDefTableEntry(pdchZnClPkOAHum, curName, 0.0, 5);
+                    PreDefTableEntry(pdchZnClPkOAMinFlow, curName, 0.0, 3);
+                    PreDefTableEntry(pdchZnClPkDOASHeatGain, curName, 0.0);
                 }
```
Comment (lgu1234) [inline src/EnergyPlus/SizingManager.cc:680]: This section reports zero values for Zone Sensible Cooling. Any character columns report "N/A", consistent with other reports.

File: src/EnergyPlus/SizingManager.cc [hunk 3]
```diff
@@ -708,6 +726,24 @@ namespace SizingManager {
...
+                    PreDefTableEntry(pdchZnHtUserDesAirFlow, curName, 0.0, 3);
+                    PreDefTableEntry(pdchZnHtDesDay, curName, "N/A");
+                    PreDefTableEntry(pdchZnHtPkTime, curName, "N/A");
+                    PreDefTableEntry(pdchZnHtPkTstatTemp, curName, 0.0);
+                    PreDefTableEntry(pdchZnHtPkIndTemp, curName, 0.0);
+                    PreDefTableEntry(pdchZnHtPkIndHum, curName, 0.0, 5);
+                    PreDefTableEntry(pdchZnHtPkOATemp, curName, 0.0);
+                    PreDefTableEntry(pdchZnHtPkOAHum, curName, 0.0, 5);
+                    PreDefTableEntry(pdchZnHtPkOAMinFlow, curName, 0.0, 3);
+                    PreDefTableEntry(pdchZnHtPkDOASHeatGain, curName, 0.0);
```
Comment (lgu1234) [inline src/EnergyPlus/SizingManager.cc]: This section reports zero values for Zone Sensible Heating. Any character columns report "N/A", consistent with other reports.

**Review by mjwitte — COMMENTED** [review]
Tested locally, solves the issue. Tested the new unit test section with develop and it fails there, runs successfully in this branch. A couple of minor cleanups and this should be good to go.

Please add a note about this to \src\Transition\OutputRulesFiles\OutputChanges9-3-0-to-9-4-0.md to alert interface developers to the change in behavior.

File: src/EnergyPlus/SizingManager.cc (hunk 1 above)
Comment (lgu1234) [reply src/EnergyPlus/SizingManager.cc]: @mjwitte No problems. My original thought was to keep the same format and may make data reading easier.

**Comment by lgu1234** [comment]
@mjwitte I am going to add a section in \src\Transition\OutputRulesFiles\OutputChanges9-3-0-to-9-4-0.md.

**Review by mjwitte — APPROVED** [review]
@lgu1234 Looks good. Once the final CI test completes, this will merge.

**Pull request merged**